.github/workflows/test.yml

```
@@ -0,0 +1,80 @@
+name: CI - Test package
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+    types: [opened, reopened, synchronize, ready_for_review]
+
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Cache python env
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.pythonLocation }}
+        key: ${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}
+    - name: Install deps
+      env:
+        PIP_UPGRADE: True
+        UPGRADE_STRATEGY: eager
+      run: |
+        python -m pip install pip flit
+        python -m flit install --deps develop
+
+    - name: Check format with black
+      run: python -m black -v --check .
+    - name: Lint with flakehell
+      run: python -m flakehell lint -v
+
+  test:
+    name: Tests
+    needs: lint
```

```suggestion
    if: github.event.pull_request.draft == false
```
I think the tests should always run, even if the linter fails.